src/FSharp.Control.TaskSeq.Test/TaskSeq.SkipWhile.Tests.fs

```diff
@@ -0,0 +1,332 @@
+            |> verifyDigitsAsString "FGHIJ" // last 4
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 5 })
+            |> verifyDigitsAsString "FGHIJ"
+    }
+
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A returns the empty sequence if always true`` variant = task {
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive ((<) -1)
+            |> verifyEmpty
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return true })
+            |> verifyEmpty
+    }
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A always skips at least the first item`` variant = task {
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive ((=) 0)
+            |> verifyDigitsAsString "BCDEFGHIJ"
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync ((=) 0 >> Task.fromResult)
+            |> verifyDigitsAsString "BCDEFGHIJ"
+    }
+
+module SideEffects =
+    [<Theory; ClassData(typeof<TestSideEffectTaskSeq>)>]
+    let ``TaskSeq-skipWhile+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 6
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T T F F F F F (stops at first F)
+        // x x x x x _ _ _ _ _ (skips exclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhile ((>) 6)
+            |> verifyDigitsAsString "FGHIJ"
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileAsync (fun x -> task { return x < 6 })
+            |> verifyDigitsAsString "FGHIJ"
+    }
+
+    [<Theory; ClassData(typeof<TestSideEffectTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 6
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T T F F F F F (stops at first F)
+        // x x x x x x _ _ _ _ (skips inclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileInclusive ((>) 6)
+            |> verifyDigitsAsString "GHIJ"
+
+        do!
+            Gen.getSeqWithSideEffect variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 6 })
+            |> verifyDigitsAsString "GHIJ"
+    }
+
+    [<Theory>]
+    [<InlineData(false, false)>]
+    [<InlineData(false, true)>]
+    [<InlineData(true, false)>]
+    [<InlineData(true, true)>]
+    let ``TaskSeq-skipWhileXXX prove it reads the entire input stream`` (inclusive, isAsync) = task {
+        let mutable x = 42 // for this test, the potential mutation should not actually occur
+        let functionToTest = getFunction inclusive isAsync ((=) 42)
```

Comment:
consider a MemberData that yields the 4 combos so they dont need to be visually scanned per test
(also maybe the MemberData can yield the function with a Case Name tag that says AsyncInclusive, SyncExclusive etc)

Reply:
I'm gonna drop the whole thing and make it inline for clarity

Reply:
wdym, 4 copies of 3/4 tests? I do like it being DRY as its very hard to review cloned code without eyes glazing over, and pretty painful to read in general

Reply:
No. Just a little function inside the test that itself takes `TaskSe.takeWhile ((=) 9999)` as an argument. It is still dry. Basically like that last test you reviewed with 4x `do`. Same idea.

DRY is not DRY if it creates too much cognitive overload.

Reply:
After doing this, the size of the single test (testing four variants) is smaller, or close to equal, but it now tests more edge cases, and is more readable. Using `InlineData` is not always an improvement, apparently :).

Reply:
Nice